### ISSUE TITLE
no-param-reassign should have option to allow reassignment in Array.prototype.reduce callbacks

### ISSUE THREAD

**Issue by mike-robertson** [eslint/eslint#8007]
It would be nice to disable this rule for certain scenarios where paramter reassignment is expected. For example, the case I find sort of annoying is that in reduce calls, the accumulator parameter often is ok to reassign as we are creating it solely for the purpose of the reduce callback. It seems wasteful to return a new object/array/whatever in each call.

**Comment by eslintbot** [comment]
Hi @mike-robertson, thanks for the issue. It looks like there's not enough information for us to know how to help you.

If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

Requesting a rule change? Please see [Proposing a Rule Change](http://eslint.org/docs/developer-guide/contributing/rule-changes) for instructions.

If it's something else, please just provide as much additional information as possible. Thanks!

**Comment by platinumazure** [comment]
This is a pretty specific exception case, so I doubt we'd want to alter the rule for this case.

You can use [comments](http://eslint.org/docs/user-guide/configuring#disabling-rules-with-inline-comments) to disable rules from being enforced on certain lines. I would recommend using those comments in this case.

**Comment by nzakas** [comment]
We'd need a full [proposal](http://eslint.org/docs/developer-guide/contributing/rule-changes) before we can really consider this, though my initial inclination is the same as @platinumazure.

**Comment by vitorbal** [comment]
Closing this issue since we haven't heard anything back for quite some time. Feel free to create a new issue if this is something you would like to discuss further.
